Feature/test

asuen39 opened the pull request.

asuen39 committed:
ブランチ移行時にコミットから漏れていたファイルのアップロード

asuen39 committed:
test画面作成

yoshizaki-takeshi commented on src/main/webapp/test/test.jsp:
```diff
@@ -0,0 +1,61 @@
+								<input type="hidden" name="questions_id" value="<%= questionsBean.getId() %>">
+							</li>
+							<li class="edit_area space_between">回答
+								<input type="text" style="width:93%;">
```
ここname属性で指定しなくて良いの？

asuen39 replied:
前の修正で既にnameは追加していました。githubに反映されていないので上げ直しました。

yoshizaki-takeshi replied:
https://github.com/asuen39/java_training/blob/b19bedf642ed3a83c0ca83e703948bb336ae701f/src/main/webapp/test/test.jsp#L47
追加されてませんね…
コミットログもあさってみたけど見当たらなかった・・・
https://github.com/asuen39/java_training/commits/b19bedf642ed3a83c0ca83e703948bb336ae701f/src/main/webapp/test/test.jsp

asuen39 replied:
Fileschangedからsrc/main/webapp/test/test.jspの44行目を確認してみるとnameがあります。

https://github.com/asuen39/java_training/pull/2/files#diff-66ee02ab2475a99779018786d158642d5aa3088278256272ed0d3692dd0502aa

yoshizaki-takeshi replied:
経緯はよくわからんけど、今のブランチの状態で直ってるからOK

yoshizaki-takeshi commented on src/main/webapp/test/test.jsp:
```diff
@@ -0,0 +1,61 @@
+						<ul>
+							<li class="edit_area space_between">
+							<!-- ループでカウントした数を設置する。 -->
+								<%= cnt %>
```
<%= i + 1 %>でも良いように思えるな。cntが余計な変数に見えてしまう。
例えばint num;と変数定義をした時点でメモリに32ビット分の容量を確保しにいくので
余計な(他で代用が効いたり、使用しなくて良いもの)変数はできるだけ減らしていきましょう。

asuen39 replied:
<%= i + 1 %>に変更しました。動作確認しました。
これだけで番号が表示できました。

yoshizaki-takeshi replied:
ここもOK

asuen39 committed:
テスト採点結果画面作成。

asuen39 committed:
テスト採点結果画面:  文言変更しました。

yoshizaki-takeshi commented on src/main/java/servlet/test/Result.java:
```diff
@@ -132,9 +132,10 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 	    	
-	    	//小数点第1位で四捨五入
+	    	//小数点第1位で四捨五入する。
+	    	//Math.roundを利用する。roundメソッドの返り値は四捨五入された値になる。
```
コメントに対しての指摘ではなくそれを使用してないことの指摘になります。
```
score = Math.round(score);
```
こんな風に四捨五入した値の結果をscoreに入れるとかしてるならわかりますが。
なので修正する前に割り切れない数字の正解数だった場合の得点表示がどうなるか確認してみてください。
7問中3問正解とか。

asuen39 replied:
書き方が古い状態です。
scoreは修正しています。再アップします。

yoshizaki-takeshi replied:
https://github.com/asuen39/java_training/blob/feature/test/src/main/java/servlet/test/Result.java#L144

まぁ、ここはOK

asuen39 committed:
修正

yoshizaki-takeshi commented on src/main/java/servlet/test/Result.java:
```diff
@@ -113,20 +113,22 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 		    		}
 		    		
+		    		//リクエストに対してcountAnswerにcount_answerを格納してセットする。
+	    			request.setAttribute("countAnswer", count_answer);
```
条件の外に出しただけで結局ループ内でやってますね。

asuen39 replied:
移動させました。

yoshizaki-takeshi replied:
https://github.com/asuen39/java_training/blob/feature/test/src/main/java/servlet/test/Result.java#L135
移動されてないよ。for文の中にいるじゃん。

asuen39 replied:
移動をさせて動作チェックしました。
問題無く処理が行われました。
勉強になります。

yoshizaki-takeshi replied:
ループの中にいても最終ループの値が上書きされてセットされるので
結果は同じではあるけどループ件数が多くなった場合、
無駄にセットされ続けるためやめといた方が無難です。

asuen39 replied:
無駄にセットされ続けてしまうんですね。気を付けます。

asuen39 committed:
コード修正

yoshizaki-takeshi commented on src/main/java/servlet/test/Result.java:
```diff
@@ -28,7 +37,133 @@ public Result() {
+	    String login_id = "";
+	    
+	    //セッションの作成
+	    HttpSession session = request.getSession();
```
request.getSession()が実行されるたびに新しいセッションが作られます。
request.getSession(false)と第1引数にfalseを指定すれば新規セッションでなく、あるものを取得してくれます。

https://www.javadrive.jp/servlet/session/index1.html

asuen39 replied:
false指定にしました。

yoshizaki-takeshi replied:
ここもOK

asuen39 committed:
historyテーブルに書き込む処理が出来たのでコミット。

asuen39 committed:
history: コメント修正

asuen39 committed:
修正

asuen39 committed:
修正

asuen39 committed:
注釈文修正

asuen39 committed:
履歴画面作成

asuen39 committed:
junit実装

asuen39 committed:
ログイン画面でのID部分合否判定実装

yoshizaki-takeshi commented on src/main/java/Dao/HistoriesDao.java:
```diff
@@ -0,0 +1,175 @@
+			
+			//sqlの実行
+			//executeUpdateで実行されたsql文の更新結果を返します。
+			st.executeUpdate();
```
更新結果を返すメソッドなので間違ってはいないけど
voidの使い方してるからこのコメントは混乱を招くかも。

asuen39 replied:
どういうコメントにすれば分かりやすいか思いつかないです。
ご教授願えないでしょうか。

yoshizaki-takeshi replied:
executeUpdateで実行されたsql文の更新、じゃダメなの？
それかコメント通りに
```
int result = st.executeUpdate();
```
ってしちゃうとか。

asuen39 replied:
ありがとうございます。上記のコメントにさせていただきました。
メモしておきます

yoshizaki-takeshi reviewed (commented):
全体通して良さそうではあるけど
CWでも言ってるコメント返信を見てから
承認(Approve)しようかと思います。

yoshizaki-takeshi commented on src/main/java/Dao/HistoriesDao.java:
```diff
@@ -0,0 +1,175 @@
+			//・変数sqlにsql文を格納する。
+			//1. historiesテーブルから全件取得する。
+			//2. レコードの取得フィールドはid、user_id、point
+			String sql = "SELECT id, user_id, point, created_at FROM histories WHERE user_id = ? ORDER BY created_at ASC";
```
ちなみにORDER BY ceated_at
でも良いすよ。省略するとascになるので。

asuen39 replied:
> SELECT id, user_id, point, created_at FROM histories WHERE user_id = ? ORDER By ceated_at

にしたえらーになりました。

yoshizaki-takeshi replied:
どういうエラーですかね・・・？
シンタックスエラー？
そうはならないはずではあるけど・・・
https://www.javadrive.jp/mysql/select/index11.html

asuen39 replied:
修正出来ました。
単純に修正ミスでした。

yoshizaki-takeshi commented on src/main/java/servlet/test/Result.java:
```diff
@@ -28,7 +38,141 @@ public Result() {
+		    			if(count_answer == 0) {
+		    				
+		    				//count_answerに0をセットする。
+		    				count_answer = 0;
```
ここって何をやってるんですかね…？
0だったら0を入れるってなんの意味があるんだろ…？

asuen39 replied:
Scoreの計算に使用する為に0をセットしています。

yoshizaki-takeshi replied:
0以外だと0はセットしないですよね…？
だったら
```
if(count_answer == 0) {
  count_answer = 0;
}
```
これ自体いらなくないですか？

asuen39 replied:
削除して動作させてみました。
問題無く動きました。無くても計算が大丈夫でした。

yoshizaki-takeshi replied:
そもそもこの処理がどうして必要なのか、が明確にならないと
また同じことやって同じようなこと指摘されちゃうんじゃないかな…

asuen39 replied:
0ってちゃんと指定しないと計算されないかなと思っていました。
メモっておきます。

yoshizaki-takeshi replied:
そういう理由だったら条件おかしくないですか？
```
if(count_answer == 0) {
```
これだと0が指定されてることが前提になっているので。
まぁ、85行目に初期値で0入れてますけど。

asuen39 replied:
ここの事を言っているんですか。

yoshizaki-takeshi replied:
違うよ。

yoshizaki-takeshi commented on src/main/java/servlet/History.java:
```diff
@@ -28,10 +35,58 @@ public History() {
+	    	//dao.findを実行する。引数にはlogin_idをセットする。
+	    	//UsersBeanの型にloginUserをセットする。
+	    	//loginUserにdao.findの結果を格納する。
+	    	UsersBean loginUser = daoUsers.find(login_id);
```
ちなみにこのlogin_idのセッションが切れてたらどうなりますか？

asuen39 replied:
エラーになります。

yoshizaki-takeshi replied:
だったらlogin_idのnullチェックいれるべきじゃないですか？
nullだったらログイン画面に飛ばす、とか。
エラーになることわかってて放置は良くないっす。

asuen39 replied:
リダイレクトの機能を追加しました。
リダイレクトでtop.javaに移動させました。
他の現場にいた時はtopページに遷移させるパターンをみた為topに移動させました。

asuen39 replied:
リダイレクト先をログイン画面に遷移に変更しました。

yoshizaki-takeshi replied:
そうですね。
ログイン後にTOPページに遷移させるのに
ログイン無視でＴＯＰに飛ばしちゃうのはまずいのでログイン画面の方が良いです。

asuen39 committed:
test.jsp 修正

asuen39 committed:
Result修正

asuen39 committed:
Result修正

asuen39 committed:
修正

asuen39 committed:
細かい修正

asuen39 committed:
リダイレクト機能の追加
1. History.javaでlogin_idが取得されなかった時にtop.javaにリダイレクトさせる。
2. top.javaにリダイレクトの値が来たらエラー文を表示させるようにする。

asuen39 committed:
コメント修正

asuen39 committed:
エラー文章を出る場所はloginに変更

asuen39 commented:
一通り確認ありがとうございました。

yoshizaki-takeshi commented on src/main/java/Dao/HistoriesDao.java:
```diff
@@ -0,0 +1,175 @@
+			st = con.prepareStatement(sql);
+			
+			//sql文内で1つ目の？にlogin_idを代入する。
+			//※setStringを使用する。
```
[NEW]
多分、コードだけ直してコメント直し忘れだと思うけど
※setStringを使用する。って書いといて↓でsetIntしてるっていうね・・・

yoshizaki-takeshi commented on src/main/java/Dao/HistoriesDao.java:
```diff
@@ -0,0 +1,175 @@
+			st = con.prepareStatement(sql);
+			
+			//sql文内で1つ目の？にlogin_idを代入する。
+			//※setStringを使用する。
```
[NEW]
ここもコメントがおかしいですね・・・

yoshizaki-takeshi commented on src/main/java/Dao/HistoriesDao.java:
```diff
@@ -0,0 +1,175 @@
+			List<HistoriesBean> list = new ArrayList<HistoriesBean>();
+			
+			//実行されたsqlからレコードが抽出される。
+			//rsをwhileで繰り返し処理を行う。nextをrsに続いて記述する。空白文字も対象にしたいから。
```
[NEW]
対象にしたい空白文字ってどのことを言ってるんでしょう…？
数値と日時しか対象にしてないけど。

yoshizaki-takeshi commented on src/main/java/servlet/AjaxServlet.java:
```diff
@@ -0,0 +1,86 @@
+	    	//listにdao.findAllの結果を格納する。
+	    	ArrayList<UsersBean> list = (ArrayList<UsersBean>) dao.findAll();
+	    	
+	    	//intの型でidMsgをnullで宣言する。
```
[NEW]
ここもコメントおかしいっすね。
intとかプリミティブ型にはnullは入れられませんよ。
Integerとかオブジェクト型なら入るけど。

yoshizaki-takeshi replied:
0か1しか使わないならboolean型でも良いような気はしなくはないけど。

yoshizaki-takeshi commented:
コードは一通り良い気はするけど
コメントが気になるとこがちらほらと・・・

asuen39 commented:
コメントと言われても。何が気になるのかよく分からないです。

yoshizaki-takeshi commented:
まだ追加のものを見てないのかな・・・？
実際のコードと違うこと書いてるとこが多いです。

asuen39 commented:
何のコメントの事を指しているか分からないです。

yoshizaki-takeshi commented:
行単位にコメントつけてるはずなんだけどな・・・
File Changedから見てみてください。

yoshizaki-takeshi commented:
一点お願いが・・・
しれっとコメント削除するのはやめてもらいたいです。
それに対しての返信コメントだけ残ってしまって
あとから流れが追えなくなるかもしれないので・・・

yoshizaki-takeshi commented:
音信不通になっちゃったんでここも見てるかわからんけど、
課題の件は終了となりました。
なのでマージしちゃってもいいすよ。
マージしたところで結局、直してないので
指摘箇所がそのままdevelopに入るだけでなんの解決にもなってないけどね。
